src/main/java/org/mamute/i18n/MamuteLocalization.java

```diff
@@ -25,27 +30,69 @@
 	protected MamuteLocalization() {
 		this(null);
 	}
-	
+
 	@Inject
 	public MamuteLocalization(HttpServletRequest request) {
 		super(request);
 		this.request = request;
 	}
-	
-	
+
+
 	@Override @Produces
 	public ResourceBundle getBundle(Locale locale) {
 		Config.set(request, Config.FMT_LOCALIZATION_CONTEXT, null);
-		ResourceBundle customBundle = super.getBundle(locale);
+		ResourceBundle customBundle = getUTF8Bundle(locale);
 		Config.set(request, Config.FMT_LOCALIZATION_CONTEXT, "mamute-messages");
-		ResourceBundle mamuteBundle = super.getBundle(locale);
-		
+		ResourceBundle mamuteBundle = getUTF8Bundle(locale);
+
 		return new MamuteResourceBundle(customBundle, mamuteBundle);
 	}
-	
+
+	public ResourceBundle getUTF8Bundle(Locale locale) {
+		Object bundle = this.findByKey("javax.servlet.jsp.jstl.fmt.localizationContext");
+		ResourceBundle unsafe = this.extractUnsafeBundle(bundle, locale);
+		return new SafeResourceBundle(unsafe);
+	}
+
+
+	private ResourceBundle extractUnsafeBundle(Object bundle, Locale locale) {
```

Comment:
I think this code is quite big and complex, I think we should extract it to a separate class to be able to unit test it easier, what do you think @khajavi?